### PR TITLE
Add support for older node versions

### PR DESCRIPTION
**Pull request by darrachequesne**
I didn't add `v0.12`, since its [LTS](https://github.com/nodejs/LTS#lts_schedule) will soon end, but it should work too :angel:

### CODE REVIEW

**Comment by leo** [comment]
Have you tested if args works on v4 without the changes you've just introduced? 😊

**Comment by darrachequesne** [comment]
It doesn't work on v4 currently because it chokes on `class` 😃

**Comment by matheuss** [comment]
I can't even `git clone ...; npm install` it on node v4 😅 
screen shot 2016-09-06 at 11 16 08 am

**Comment by leo** [comment]
@matheuss That's actually just because on node older than 6, gulp is not able to load `babel-register` from `gulp-babel`. So it needs to be a separate dependency. But I don't want that, because it's a lot of code for a small functionality. 😄

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 language: node_js
 sudo: false
-node_js: node
+node_js:
+  - node
+  - "6"
```
Comment (leo) [inline .travis.yml]: Please remove the `"`s.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+  - "4"
```
Comment (leo) [inline .travis.yml]: Please remove the `"`s.

File: package.json [hunk 3]
```diff
@@ -13,6 +13,9 @@
     "test": "xo && ava"
   },
   "babel": {
+    "presets": [
+      "es2015"
+    ],
     "plugins": [
       "add-module-exports",
       "transform-es2015-modules-commonjs"
```
Comment (leo) [inline package.json]: If you really want to use the preset, this isn't needed anymore.

File: package.json [hunk 4]
```diff
@@ -61,8 +64,10 @@
   "homepage": "https://github.com/leo/args#readme",
   "devDependencies": {
     "ava": "^0.16.0",
+    "babel-core": "^6.14.0",
```
Comment (leo) [inline package.json]: Why? 😄

**Comment by leo** [comment]
All in all, I'm not fully sure if I want to support v4 again. I really want to push the community forward to using v6... Hmmm...

**Comment by matheuss** [comment]
- I can't imagine a reason for not using node @ latest
- `args` is used in CLI tools, so whoever is using it know what's going on with node 

Given that, I agree on keeping the support limited to `v6`. There's only advantages on that :smile:

File: package.json (hunk 4 above)
Comment (darrachequesne) [inline package.json]: Without that, I got:
```
$ gulp -v
[22:39:46] Failed to load external module babel-register
[22:39:46] Failed to load external module babel-core/register
[22:39:46] Failed to load external module babel/register
[22:39:46] CLI version 3.9.1
[22:39:46] Local version 3.9.1
```

**Comment by darrachequesne** [comment]
@leo @matheuss I totally get your point :+1:

Sorry for the noise!

**Comment by leo** [comment]
@darrachequesne No problem at all! Thank you for your effort nevertheless.

**Pull request closed without merging**